File: ohce.ts
import readline from 'readline';

// Create readline interface
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

// Function to check if a string is a palindrome
function isPalindrome(str: string): boolean {
  return str === str.split('').reverse().join('');
}

// Start the application
console.log('Bonjour');

rl.on('line', (input: string) => {
  if (input === 'exit') {
    console.log('Au revoir');
    rl.close();
  } else {
    console.log(`Mirror: ${input}`); // Mirror feature
    if (isPalindrome(input.replace(/\s+/g, '').toLowerCase())) {
      console.log('Bien dit'); // Palindrome response
    }
  }
});

File: ohce.py
def is_palindrome(s):
    return s == s[::-1]

def main():
    print("Bonjour")

    while True:
        user_input = input("Entrez un texte (ou 'quitter' pour arrêter) : ")

        if user_input.lower() == 'quitter':
            print("Au revoir")
            break

        print(f"Vous avez écrit : {user_input}")

        if is_palindrome(user_input.replace(" ", "").lower()):
            print("Bien dit")

if __name__ == "__main__":
    main()
